R/queue.R

```diff
@@ -62,6 +64,29 @@ Queue <- R6::R6Class("Queue", #nolint
       rrq::rrq_worker_len(self$controller)
     },
 
+    #' @description
+    #' Gets status of packet run
+    #'
+    #' @param job_id Id of redis queue job to get status of.
+    #' @return status of redis queue job
+    get_status = function(job_id) {
+      if (!rrq::rrq_task_exists(job_id, controller = self$controller)) {
+        porcelain::porcelain_stop("Job ID does not exist")
+      }
+      status <- rrq::rrq_task_status(job_id, controller = self$controller)
+      times <- rrq::rrq_task_times(job_id, controller = self$controller)
+
+      list(
+        status = scalar(status),
+        queue_position = if (status == "PENDING") scalar(rrq::rrq_task_position(job_id, controller = self$controller)) else NULL,
+        time_queued = scalar(times[1]),
+        time_started = if (!is.na(times[2])) scalar(times[2]) else NULL,
+        time_complete = if (!is.na(times[3])) scalar(times[3]) else NULL,
+        packet_id = if (status == "COMPLETE") scalar(rrq::rrq_task_result(job_id, controller = self$controller)) else NULL,
```

Comment:
i think we should also handle the error case here, a task status can also be "ERROR", i found all the statuses here: https://github.com/mrc-ide/rrq/blob/ba2784057e8ad83e304ac9ae44081110c6b3d19c/R/common.R

for now we can ignore the other statuses